### DISCUSSION TITLE
Use `lp_terminal_format` out of the prompt

### DISCUSSION THREAD

**Discussion by nojhan** [liquidprompt/liquidprompt#783] [Q&A] [answered]
Using [lp_terminal_format](https://liquidprompt.readthedocs.io/en/stable/functions/public.html#lp_terminal_format) obviously works when building up the prompt within a theme. Now, I would want to use it, but out in a function called by the user.

However, when I do that and print the returned string, I get escape codes, e.g.: `\[\] example \[\]`

Is there a way to use `lp_terminal_format` (along with `$NO_COL`) in a free function?

### Minimal working example

```sh
_lp_escaped_theme_activate() {
    _lp_default_theme_activate
}
_lp_escaped_theme_directory() {
    _lp_default_theme_directory
}
_lp_escaped_theme_prompt() {
    lp_terminal_format 0 196 1
    PS1="${lp_terminal_format}>${NO_COL}"
}
user_function() {
    local n=$'\n'
    lp_terminal_format 0 196 1
    printf '%s' "${lp_terminal_format} example ${NO_COL}${n}"
}
```

![Capture d’écran du 2023-07-10 10-11-09](https://github.com/nojhan/liquidprompt/assets/285009/3c77d6d0-47e3-40b1-b726-29241f9ad1c0)

**Comment by Rycieos** [comment, marked as answer]
You could manually remove the escape codes:
```bash
user_function() {
    local lp_terminal_format string
    lp_terminal_format 0 196 1
    string="${lp_terminal_format} example ${NO_COL}"
    string="${string//"$_LP_OPEN_ESC"}"
    string="${string//"$_LP_CLOSE_ESC"}"
    printf '%s\n' "$string"
}
```
It uses private Liquid Prompt variables, so it could break at some point. If this is something others might want to do, we could probably come up with a way to support this.
